docs/source/users/problem_definition_files/spinw.rst

```diff
@@ -15,105 +32,132 @@ software, name, description
+    mf = multifit(w);
+    mf = mf.set_fun(forefunc);
+    mf = mf.set_pin (pin);
+    [wout,fitpar] = mf.simulate();
+    [ ~, ~, msk] = sigvar_get(wout);
+  
+  The last two lines of the wye_function applies the `msk` on the `y` and `e` data. As the `e` from retrieved above is the
+  variance we have squared root the value to get the standard deviation.
 
-mc_points
-  SpinW used Monte Carlo sampling during it's calculation. This controls the
-  number of samples taken. It is left to the experienced SpinW user to tune
-  this.
+  .. code-block:: rst
 
-  e.g.::
-    mc_points = 5
+    y = spinw_y(msk);
+    e = sqrt(spinw_e(msk));
 
-intrinsic_energy_width
-  Each call to SpinW also requires an energy width. This parameter is not
-  fitted.
+  Examples of the wye_function:
 
-  e.g.::
-    intrinsic_energy_width = 0.1
+  .. literalinclude:: ../../../../examples/benchmark_problems/SpinW/m_scripts/wye_functions/fb_wye_IX_1D_test1.m
+  
+  .. literalinclude:: ../../../../examples/benchmark_problems/SpinW/m_scripts/wye_functions/fb_wye_pcsmo_test.m
+
+simulate_function
+  The simulate_function is defined by a matlab file which returns the derived y values from `simulate()` for the fitting function.
+  This matlab file takes in the `w`, `fitpars` (fitting parameters) and `msk`. The `w` and `msk` are the same as the 
+  wye_function. The `fitpars` are determined by the current minimizer.  
+
+Explained Example of the simulate_function:
+
+.. code-block:: rst
+  
+  forefunc = @mftest_gauss_bkgd;
+  mf = multifit(w);
+  mf = mf.set_fun(forefunc);
+  mf = mf.set_pin(fitpars);
+  [wout,fitpar] = mf.simulate();
+  [spinw_y, e] = sigvar_get(wout);
+  spinw_y=spinw_y(msk);
+
+.. note:: 
+  If the benchmark problem is `tobyfit` or uses monte carlo. A persisent seed needs to be set before simulate is ran.
+  This make sure that it uses the same seed everytime :code:`simulate()` is ran.  
+    
+  .. code-block:: rst
+
+    persistent seed
+    if isempty (seed)
+        rng(3,"twister");
+        seed = rng();
+    else 
+        rng(seed);
+    end
+
+.. note:: 
+  If the SpinW benchmark problem is run in parallel make sure to turn off hpc after :code:`simulate()` in the simulate_function 
+  matlab script. 
+
+  .. code-block:: rst
+
+    hpc('off')
+
+
+Examples of the simulate_function:
+  
+.. literalinclude:: ../../../../examples/benchmark_problems/SpinW/m_scripts/simulate_functions/fb_simulate_IX_1D_test1.m
 
-spinwpars
-  SpinW models have some additional parameters which are passed to TobyFit.
-  These can be set for the model by adding them to the file as a new value
-  prefixed with "spinwpars_".
+.. literalinclude:: ../../../../examples/benchmark_problems/SpinW/m_scripts/simulate_functions/fb_simulate_pcsmo_test.m
 
-  It is left to the experienced SpinW user to know what their problem requires
-  in this regard.
+.. note::
+   All the functions needed in the fitting must be in the subdirectory of the benchmark problem.
```

Comment:
Is this due to absolute vs. relative pathing or something more fundamental?

Reply:
In the SpinW parser we add the benchmark problem directory to the path. If the functions needed for fitting are not global functions in matlab we would expected it to be defined within the sub directory of the benchmark problem. If it is not the run will fail, stating that the functions can't be found.